tests/lava/lib/peripherals/dvs/test_prophesee.py

```diff
@@ -0,0 +1,343 @@
+                filename=SEQUENCE_FILENAME_RAW,
+                sensor_shape=(height, width),
+                max_events_per_dt=max_events_per_dt,
+            )
+
+        with self.assertRaises(ValueError):
+            PropheseeCamera(
+                filename=SEQUENCE_FILENAME_RAW,
+                sensor_shape=(height, width),
+                num_output_time_bins=num_output_time_bins,
+            )
+
+        with self.assertRaises(ValueError):
+            PropheseeCamera(
+                filename=SEQUENCE_FILENAME_RAW,
+                sensor_shape=(height, width),
+                biases=biases,
+            )
+
+
+class TestPyPropheseeCameraModel(unittest.TestCase):
+    def test_init(self):
+        """Test that the PyPropheseeCameraModel ProcessModel is instantiated
+        correctly."""
+
+        reader = RawReader(SEQUENCE_FILENAME_RAW)
+        height, width = reader.get_size()
+        del reader
+
+        transformations = Compose(
+            [
+                Downsample(factor=0.5),
+            ]
+        )
+        num_output_time_bins = 2
+
+        proc_params = {
+            "shape": (num_output_time_bins, 2, height, width),
+            "filename": SEQUENCE_FILENAME_RAW,
+            "biases": None,
+            "filters": [
+                ActivityNoiseFilterAlgorithm(
+                    width=width, height=height, threshold=1000
+                )
+            ],
+            "max_events_per_dt": 10**8,
+            "transformations": transformations,
+            "num_output_time_bins": num_output_time_bins,
+        }
+
+        pm = PyPropheseeCameraModel(proc_params)
+
+        self.assertIsInstance(pm, PyPropheseeCameraModel)
+        self.assertIsInstance(pm.reader, RawReader)
+
+    def test_base_functionality_file(self):
+        """Test that running a PropheseeCamera works using a data file."""
+        reader = RawReader(SEQUENCE_FILENAME_RAW)
+        height, width = reader.get_size()
+        del reader
+
+        num_steps = 2
+
+        camera = PropheseeCamera(
+            filename=SEQUENCE_FILENAME_RAW, sensor_shape=(height, width)
+        )
+
+        run_condition = RunSteps(num_steps=num_steps)
+        run_cfg = Loihi2SimCfg()
+        camera.run(condition=run_condition, run_cfg=run_cfg)
+        camera.stop()
+
+    def test_base_functionality_dat_file(self):
+        """Test that running a PropheseeCamera works using a dat data file."""
+        # The DAT file should have the same resolution as the RAW file
+        reader = RawReader(SEQUENCE_FILENAME_RAW)
+        height, width = reader.get_size()
+        del reader
+
+        num_steps = 2
+
+        camera = PropheseeCamera(
+            filename=SEQUENCE_FILENAME_DAT, sensor_shape=(height, width)
+        )
+
+        run_condition = RunSteps(num_steps=num_steps)
+        run_cfg = Loihi2SimCfg()
+        camera.run(condition=run_condition, run_cfg=run_cfg)
+        camera.stop()
+
+    @unittest.skip("Needs live camera")
```

Comment:
Ideally, this skip should be gated by a bool variable that checks whether or not there is a Prophesee camera connected. Otherwise this test is more or less useless as it will never be used.